src/components/tabs/FunctionsTab.vue

```diff
@@ -76,6 +80,24 @@ export default class FunctionsTab extends Vue {
   private selectedFile = -1;
   private selectedFunction = -1;
 
+  /**
+   * Watches the rendering of the CodeVault in order to update selected function.
+   */
+  @Watch('inCodeVault')
```

Comment:
Couldn't you just watch `nodeTriggeringCodeVault` instead? Since otherwise the `FunctionsTab` isn't displayed anyway so it doesn't matter

Reply:
Well it could be that you entered the Code Vault from the Navbar right? And in that case you probably want to not have anything selected?

Reply:
Which makes me think that there's a bug here aha

Reply:
So quick recap here: I've added the mutation `unlinkNode` to be called when switching to the Code Vault from the Navbar (which was previously here as we had `state.nodeTriggeringCodeVault = undefined` in the `enterCodeVault` editor). That way, if you go to the Code Vault from the Navbar, the selection is cleared, and the IDE is as well. I think that's the expected behaviour. What do you think @tomproberts @gidonkatten?

Reply:
Yeah I think that's good